lib/wasi/src/runtime/resolver/cache.rs

```diff
@@ -0,0 +1,228 @@
+use std::{sync::Mutex, time::Instant};
+
+use crate::{
+    bin_factory::BinaryPackage,
+    http::HttpClient,
+    runtime::resolver::{PackageResolver, ResolverError, WebcIdentifier},
+};
+
+/// A resolver that wraps a [`PackageResolver`] in an in-memory LRU cache.
+#[derive(Debug)]
+pub struct InMemoryCache<R> {
+    resolver: R,
+    packages: Mutex<Vec<CacheEntry>>,
+    config: CacheConfig,
+}
```

Comment:
Why it needs to be LRU? If so, why do we need to implement us the LRU instead of reusing other package?

Reply:
The idea was so we can start adding cache invalidation to Deploy because at the moment, all modules and `BinaryPackage`s that end up in an in-memory cache will stay there until the program exits.

In retrospect, it's probably an overcomplication at the moment. I think I'll rip the LRU stuff out and use a `RwLock<HashMap>` that never gets cleaned up, just like what `ModuleCache` did before it.